packages/c/catch2/xmake.lua

```diff
@@ -19,6 +19,7 @@ package("catch2")
     add_versions("v2.9.2", "dc486300de22b0d36ddba1705abb07b9e5780639d824ba172ddf7062b2a1bf8f")
 
     add_configs("cxx17", {description = "Compiles Catch as a C++17 library (requires a C++17 compiler).", default = true, type = "boolean"})
+    add_configs("use_catch2main",{description = "Use Catch2 entry point", default = true, type = "boolean"})
```

Comment:
please rename to `catch2main`

Reply:
This should be removed and replaced by components

Reply:
Yeah, but libsdl has a config for compatibility purposes because it existed before components, if you introduce this you don't have to add a config

Reply:
The use of config is indeed optional, and we can disable catch2main in the following way 

```lua
add_requires("catch2")
target("test")
    -- ..
    add_packages("catch2", {components = "lib"})
```